hazelcast/src/main/java/com/hazelcast/nio/tcp/TcpIpConnectionManager.java

```diff
@@ -129,7 +130,8 @@ public TcpIpConnectionMonitor createNew(Address endpoint) {
     @Probe
     private final MwCounter closedCount = newMwCounter();
 
-    private final ScheduledExecutorService scheduler = new ScheduledThreadPoolExecutor(4);
+    private final ScheduledExecutorService scheduler
+            = new ScheduledThreadPoolExecutor(4, new ThreadFactoryImpl("TcpIpConnectionManager-thread-"));
```

Comment:
There will be a quite a big set of changes required because now there is going to be a reliance of the HazelcastThreadGroup. Not only on the server, but also on the client.